api/server/server.go

```diff
@@ -1160,9 +1164,11 @@ func postBuild(eng *engine.Engine, version version.Version, w http.ResponseWrite
 		authConfig        = &registry.AuthConfig{}
 		configFileEncoded = r.Header.Get("X-Registry-Config")
 		configFile        = &registry.ConfigFile{}
-		job               = eng.Job("build")
+		job               = builder.NewBuildConfig(eng.Logging, eng.Stderr)
 	)
 
+	b := &builder.BuilderJob{eng, getDaemon(eng)}
```

Comment:
Oh I see now what's going on.  I'm still not sure why we need to create a 'b' object, why not just make Build() a global func and pass in the engine/daemon as an arg or in the BuildConfig ?  Or even make Build a func on Daemon.